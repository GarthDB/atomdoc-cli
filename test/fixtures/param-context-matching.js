/**
 *  Public: simple sample funtion.
 *
 *  * `arguments` {Object}
 *    * `str` {String} the name
 *    * `val` (optional) {Number} some number
 *
 *  ## Examples
 *
 *  ```js
 *  h({ str: "bar", val: 42 });
 *  ```
 */
function h ({ str, val = 12 }) {
  console.log(str, val)
}

/**
 *  Public: simple sample funtion.
 *
 *  * `arguments` {Object}
 *    * `name` {String} the name
 *    * `val` (optional) {Number} some number
 *
 *  ## Examples
 *
 *  ```js
 *  h({ str: "bar", val: 42 });
 *  ```
 */
function g ({ str, val = 12 }) {
  console.log(str, val)
}

/**
 *  Public: simple sample funtion.
 *
 *  * `arguments` {Object}
 *    * `str` {String} the name
 *    * `val` {Number} some number
 *
 *  ## Examples
 *
 *  ```js
 *  h({ str: "bar", val: 42 });
 *  ```
 */
function i ({ str, val = 12 }) {
  console.log(str, val)
}
