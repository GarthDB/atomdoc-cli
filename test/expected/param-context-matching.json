[
  {
    "type": "FunctionDeclaration",
    "name": "h",
    "className": false,
    "start": 223,
    "end": 281,
    "definitionLine": 14,
    "args": [
      {
        "name": "arguments",
        "optional": false,
        "children": [
          {
            "name": "str",
            "optional": false
          },
          {
            "name": "val",
            "optional": true
          }
        ]
      }
    ],
    "returns": []
  },
  {
    "type": "FunctionDeclaration",
    "name": "g",
    "className": false,
    "start": 507,
    "end": 565,
    "definitionLine": 31,
    "args": [
      {
        "name": "arguments",
        "optional": false,
        "children": [
          {
            "name": "str",
            "optional": false
          },
          {
            "name": "val",
            "optional": true
          }
        ]
      }
    ],
    "returns": []
  },
  {
    "type": "FunctionDeclaration",
    "name": "i",
    "className": false,
    "start": 779,
    "end": 837,
    "definitionLine": 48,
    "args": [
      {
        "name": "arguments",
        "optional": false,
        "children": [
          {
            "name": "str",
            "optional": false
          },
          {
            "name": "val",
            "optional": true
          }
        ]
      }
    ],
    "returns": []
  },
  {
    "type": "FunctionDeclaration",
    "name": "j",
    "className": false,
    "start": 1015,
    "end": 1073,
    "definitionLine": 63,
    "args": [
      {
        "name": "arguments",
        "optional": false,
        "children": [
          {
            "name": "str",
            "optional": false
          },
          {
            "name": "val",
            "optional": true
          }
        ]
      }
    ],
    "returns": []
  }
]
